tests/test_numbers.py

```diff
@@ -283,11 +292,51 @@ def test_format_decimal():
     assert numbers.format_decimal(1.2345, locale='sv_SE') == u'1,234'
     assert numbers.format_decimal(1.2345, locale='de') == u'1,234'
     assert numbers.format_decimal(12345.5, locale='en_US') == u'12,345.5'
+    assert numbers.format_decimal(0001.2345000, locale='en_US') == u'1.234'
+    assert numbers.format_decimal(-0001.2346000, locale='en_US') == u'-1.235'
+    assert numbers.format_decimal(0000000.5, locale='en_US') == u'0.5'
+    assert numbers.format_decimal(000, locale='en_US') == u'0'
+
+
+def test_format_decimal_quantization():
+    # Test precision conservation.
+    test_data = [
```

Comment:
You can use [`@pytest.mark.parametrize`](https://docs.pytest.org/en/latest/parametrize.html) here:

```python
@pytest.mark.parametrize('input_value, expected', [
 ('10000', '10,000'),
 # (etc, etc.)
])
def test_format_decimal_quantization(input_value, expected):
  # ...
```

This has the advantage that each of these cases will show up (and be addressable) as a separate test case, so if one of those fails, the reader will not have to read the locals to figure out which particular case failed. :)